navigator.go

```diff
@@ -99,8 +100,7 @@ func (n navigator) Follow(rel string) navigator {
 // Followf adds a relation to the follow queue of the navigator, with a
 // set of parameters to expand on execution.
 func (n navigator) Followf(rel string, params P) navigator {
-	relations := make([]relation, 0, len(n.path)+1)
-	copy(n.path, relations)
```

Comment:
The known capacity is more efficient. The first argument to [copy](http://golang.org/pkg/builtin/#copy) is the destination. Swapping the arguments should also fix.